tools/dotnet-linker/Steps/ComputeAOTArguments.cs

```diff
@@ -19,6 +19,8 @@ protected override void TryEndProcess ()
 
 			var app = Configuration.Application;
 			var outputDirectory = Configuration.AOTOutputDirectory;
+			var dedupFileName = Path.GetFileName (Configuration.DedupAssembly);
+			var isDedupEnabled = Configuration.Target.Assemblies.Any (asm => Path.GetFileName (asm.FullPath) == dedupFileName);
```

Comment:
Same as above. Using item metadata instead to prevent comparing against a filename.

Reply:
> Anyway, we plan to simplify the dedup build process, so such assembly shouldn't be required soon.

That's great, it would make most of this PR unnecessary!

Reply:
Correct, it was allowed by https://github.com/dotnet/runtime/pull/83711. As it would require runtime changes and additional testing, I suggest doing it as a follow-up effort on both runtime and xamarin repositories, as outlined in https://github.com/dotnet/runtime/issues/83973#issue-1642051395.

Reply:
I was wondering whether we could remove the dependency on the assembly filename and that we could do something like:
```xml
<ItemGroup>
    <ManagedAssemblyToLink Include="$(_DedupAssembly)" />
	    <IsDedupAssembly>true</IsDedupAssembly>
    <ManagedAssemblyToLink>
</ItemGroup>
```
but I missed the fact that the custom linker step does not have access to MSBuild item metadata at this point. My bad, and thanks for checking this out.